src/Models/UserIdentityModel.php

```diff
@@ -211,6 +214,142 @@ public function getAllAccessTokens(User $user): array
             ->findAll();
     }
 
+    // HMAC
+    /**
+     * Find and Retrieve the HMAC AccessToken based on Token alone
+     *
+     * @return ?AccessToken Full HMAC Access Token object
+     */
+    public function getHmacTokenByKey(string $key): ?AccessToken
+    {
+        return $this
+            ->where('type', HmacSha256::ID_TYPE_HMAC_TOKEN)
+            ->where('secret', $key)
+            ->asObject(AccessToken::class)
+            ->first();
+    }
+
+    /**
+     * Generates a new personal access token for the user.
+     *
+     * @param string   $name   Token name
+     * @param string[] $scopes Permissions the token grants
+     *
+     * @throws Exception
+     * @throws ReflectionException
+     */
+    public function generateHmacToken(User $user, string $name, array $scopes = ['*']): AccessToken
+    {
+        $this->checkUserId($user);
+
+        //        helper('text');
```

Comment:
Uncomment the line.
Otherwise `random_byte()` cannot be used.

Reply:
There is no `random_bytes()` method in the text helper.  `random_bytes()` is native to PHP.  I will remove the commented out line.